core/arch/aarch64/proc.c

```diff
@@ -74,6 +74,12 @@ proc_num_simd_saved(void)
     return num_simd_saved;
 }
 
+void
+proc_set_num_simd_saved(int num)
+{
+    num_simd_saved = num;
```

Comment:
Despite the race question for the unprotect call above, I don't see a malicious race here. Yes it is racy, but it is a latch, i.e. whoever comes first will write (16 becomes 32). Everybody thread that is running into AVX-512 code is guaranteed to have this updated before any context switch. WDYT?

But - proc_num_simd_saved() will end up being an API function that is provided just for informal reasons. The same information can be derived from dr_mcontext_zmm_fields_valid(). Therefore, 

Option 1) remove proc_num_simd_saved(). It is not needed internally AFAI can see. This is not my favorite option because a) we announced it in the API and b) it is nice to have it.

Option 2) move num_simd_saved into a different unprotected section. I don't favor this option either because the write happens only once, and while it is not used internally and not a security issue, it may be one if a client uses it.

Option 3) Leave as is and unprotect for the one-time write <= my favorite option. Still have the question above about the mutex.

Reply:
Yes, option 3, and with an ATOMIC_* method for the store esp on this non-x86-memory-model platform.

Reply:
Done.